lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -0,0 +1,386 @@
+        params['authMessage'] = ''
+        params['authOnly'] = false
+        params['availableAuthAmount'] = 0
+        params['batch'] = nil
+        params['batchId'] = nil
+      end
+
+      def add_credit_card(params, credit_card, action, options)
+        return unless credit_card
+
+        params['cardAccount'] ||= {}
+        card_details = params['cardAccount'] = {}
+
+        case action
+        when 'purchase'
+          card_details['avsStreet'] = options[:avsStreet]
+          card_details['avsZip'] = options[:avsZip]
+          card_details['cvv'] = credit_card.verification_value
+          card_details['entryMode'] = 'Keyed'
+          card_details['expiryDate'] = expdate(credit_card)
+          card_details['expiryMonth'] = format(credit_card.month, :two_digits).to_s
+          card_details['expiryYear'] = format(credit_card.year, :two_digits).to_s
+          card_details['last4'] = nil
+          card_details['magstripe'] = nil
+          card_details['number'] = credit_card.number
+        when 'refund'
+          card_details['cardId'] = credit_card['cardId']
+          card_details['cardPresent'] = credit_card['cardPresent']
+          card_details['cardType'] = credit_card['cardType']
+          card_details['entryMode'] = credit_card['entryMode']
+          card_details['expiryMonth'] = credit_card['expiryMonth']
+          card_details['expiryYear'] = credit_card['expiryYear']
+          card_details['hasContract'] = credit_card['hasContract']
+          card_details['isCorp'] = credit_card['isCorp']
+          card_details['isDebit'] = credit_card['isDebit']
+          card_details['last4'] = credit_card['last4']
+          card_details['token'] = credit_card['token']
+        end
+      end
+
+      def expdate(credit_card)
+        "#{format(credit_card.month, :two_digits)}/#{format(credit_card.year, :two_digits)}"
+      end
+
+      @request_params = {
+        achIndicator: nil,
+        authCode: nil,
+        authOnly: false,
+        bankAccount: nil,
+        cardPresent: false,
+        cardPresentType: 'CardNotPresent',
+        isAuth: true,
+        is_settle_funds: true,
+        isTicket: false,
+        merchantId: 514_391_592,
+        mxAdvantageEnabled: false,
+        mxAdvantageFeeLabel: '',
+        paymentType: 'Sale',
+        purchases: [{ taxRate: '0.0000', additionalTaxRate: nil, discountRate: nil }],
+        shouldGetCreditCardLevel: true,
+        shouldVaultCard: true,
+        source: 'QuickPay',
+        sourceZip: '94102',
+        taxExempt: false,
+        tenderType: 'Card',
+        terminals: []
+      }
+      def purchases
+        [{ taxRate: '0.0000', additionalTaxRate: nil, discountRate: nil }]
+      end
+
+      def add_type_merchant_purchase(params, merchant, is_settle_funds, options)
+        params['cardPresent'] = false
+        params['cardPresentType'] = 'CardNotPresent'
+        params['isAuth'] = true
+        params['is_settle_funds'] = is_settle_funds
```

Comment:
@mikebyrneprth is PPS expecting `is_settle_funds` or are they expecting `isSettleFunds`? If the latter, please update the areas in the code that will go to PPS in the `def commit` but leave anything in Active Merchant from Spreedly as `is_settle_funds` for Ruby convention